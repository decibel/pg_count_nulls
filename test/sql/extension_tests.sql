\set ECHO none

\set schema public
\i test/helpers/setup.sql

\set schema schema_to_load_count_nulls
CREATE SCHEMA :schema;
ALTER EXTENSION count_nulls SET SCHEMA :schema;

\i test/core/functions.sql

CREATE FUNCTION shutdown__drop_all
() RETURNS SETOF text LANGUAGE plpgsql AS $body$
BEGIN
    RETURN NEXT lives_ok(
        $$DROP EXTENSION count_nulls$$
    );

    RETURN NEXT lives_ok(
        $$DROP SCHEMA schema_to_load_count_nulls$$
    );
END
$body$;

--SET client_min_messages = debug;

SELECT * FROM runtests( '_null_count_test'::name );
