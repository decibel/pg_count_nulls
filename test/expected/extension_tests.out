\set ECHO none
    # Subtest: _null_count_test.test__definition()
    ok 1 - Function null_count(anyarray) should return bigint
    ok 2 - Function null_count(anyarray) should not be strict
    ok 3 - Function null_count(anyarray) should be IMMUTABLE
    ok 4 - Function null_count(json) should return bigint
    ok 5 - Function null_count(json) should not be strict
    ok 6 - Function null_count(json) should be IMMUTABLE
    ok 7 - Function not_null_count(anyarray) should return bigint
    ok 8 - Function not_null_count(anyarray) should not be strict
    ok 9 - Function not_null_count(anyarray) should be IMMUTABLE
    ok 10 - Function not_null_count(json) should return bigint
    ok 11 - Function not_null_count(json) should not be strict
    ok 12 - Function not_null_count(json) should be IMMUTABLE
    1..12
ok 1 - _null_count_test.test__definition
    # Subtest: _null_count_test.test__functionality()
    ok 1 - Test null_count(a, b, c)
    ok 2 - Test null_count(a, b, c)
    1..2
ok 2 - _null_count_test.test__functionality
ok 3
ok 4
1..4
